### PR TITLE
Pull in the new Fulcio client

### PR DESCRIPTION
**Pull request by mattmoor**
Related to: https://github.com/sigstore/fulcio/pull/262

Signed-off-by: Matt Moore <mattmoor@chainguard.dev>

#### Release Note
```release-note

```

### CODE REVIEW

**Comment by mattmoor** [comment]
This is really just for presubmits.  I am going to move the client code into my Fulcio change, but it was faster to try things doing it here.

**Comment by mattmoor** [comment]
Alright, the Fulcio change landed, so this is RFAL!

File: cmd/cosign/cli/fulcio/fulcio.go [hunk 1]
```diff
@@ -69,66 +55,37 @@ func (rf *realConnector) OIDConnect(url, clientID, secret string) (*oauthflow.OI
...
-	params.SetCertificateRequest(
-		&models.CertificateRequest{
-			PublicKey: &models.CertificateRequestPublicKey{
-				Algorithm: models.CertificateRequestPublicKeyAlgorithmEcdsa,
-				Content:   &content,
-			},
-			SignedEmailAddress: &signedChallenge,
+	cr := api.CertificateRequest{
+		PublicKey: api.Key{
+			Algorithm: "ecdsa",
```
Comment (dekkagaijin) [inline cmd/cosign/cli/fulcio/fulcio.go:78]: should make this a custom type and/or enum imho

**Review by dekkagaijin — APPROVED** [review]

File: cmd/cosign/cli/fulcio/fulcio.go (hunk 1 above)
Comment (mattmoor) [reply cmd/cosign/cli/fulcio/fulcio.go:78]: Want to send a PR to Fulcio?

Comment (dekkagaijin) [reply cmd/cosign/cli/fulcio/fulcio.go:78]: will do

**Pull request merged**